src/lib/atoms/Rating/Rating.tsx

```diff
@@ -0,0 +1,321 @@
+     */
+    character?: FC | string | number | ((i: number) => ReactNode);
+
+    /**
+     * The color is used for the rating elements when hovered over or selected.
+     */
+    color?: string;
+
+    /**
+     * The background color of the rating elements.
+     */
+    bgColor?: string;
+
+    /**
+     * Allows users to select half values in the rating component, enabling finer granularity.<br>
+     * For example, if `count` is 5, users can select 1, 1.5, 2, 2.5, and so on up to 5.
+     */
+    halfAllow?: boolean;
+
+    /**
+     * When set to `true`, the rating component becomes read-only, preventing any interaction.
+     */
+    readonly?: boolean;
+
+    /**
+     * The size of the rating elements.<br>
+     * Possible values: `small | medium | big`
+     */
+    size?: 'small' | 'medium' | 'big';
+
+    /**
+     * Callback function that is called when the rating value changes.<br>
+     * Receives the new rating value as an argument.
+     */
+    onChange?: (rating: number) => void;
+}
+
+interface IElementProps {
+    isIcon?: boolean;
+    Element: JSX.Element | string | number;
+    compareElement?: boolean;
+    color: string;
+    bgColor: string;
+    size: string;
+}
+
+const Element: FC<IElementProps> = ({ isIcon = false, Element, compareElement = false, color, bgColor, size }) => {
+    const currentColor = compareElement ? color : bgColor;
+
+    const className = classNames('rating__icon', `s-${size}`);
+
+    const currentSizes = {
+        width: sizes[size],
+        height: sizes[size]
+    };
+
+    const elementProps = {
+        fill: currentColor,
+        ...currentSizes,
+        style: { color: currentColor, ...currentSizes }
+    };
+
+    return isIcon ? (
+        cloneElement(Element as JSX.Element, {
+            ...elementProps,
+            className
+        })
+    ) : (
+        <div {...elementProps} className={className}>
+            {Element}
+        </div>
+    );
+};
+
+const Rating: FC<IRatingProps> = (props) => {
+    const {
+        defaultValue = 0,
+        count = 5,
+        onChange,
+        character = SvgSquareIcon,
+        color = 'var(--hero)',
+        bgColor = 'rgba(var(--hero-rgb), 0.3)',
+        halfAllow = false,
+        readonly = false,
+        size = 'small',
+        value,
+        ...restProps
+    } = props;
+
+    const isControlled = 'value' in props;
+    const isDefaultValueExist = 'defaultValue' in props;
+
+    const isRTLMode = document.dir === 'rtl';
+    const currentValue = value || defaultValue || 0;
+
+    const [rating, setRating] = useState(currentValue);
+    const [hoveredValue, setHoveredValue] = useState(0);
+    const [regardingPosition, setRegardingPosition] = useState(0);
+    const [remainingRating, setRemainingRating] = useState(0);
+    const [temporaryRating, setTemporaryRating] = useState(0);
+    const [disableMouseMovie, setDisableMouseMovie] = useState(false);
```

Comment:
Please fix the typo and refactor on used places. 

```suggestion
    const [disableMouseMove, setDisableMouseMove] = useState(false);
```